app/services/cypher_generator.py

```diff
@@ -0,0 +1,170 @@
+from typing import List
+import logging
+from dotenv import load_dotenv
+import neo4j
+from app.services.query_generator_interface import QueryGeneratorInterface
+from neo4j import GraphDatabase
+import glob
+import os
+import json
+from neo4j.graph import Node, Relationship
+
+load_dotenv()
+
+class CypherQueryGenerator(QueryGeneratorInterface):
+    def __init__(self, dataset_path: str):
+        self.driver = GraphDatabase.driver(
+            os.getenv('NEO4J_URI'),
+            auth=(os.getenv('NEO4J_USERNAME'), os.getenv('NEO4J_PASSWORD'))
+        )
+        # self.dataset_path = dataset_path
+        # self.load_dataset(self.dataset_path)
+
+    def close(self):
+        self.driver.close()
+
+    def load_dataset(self, path: str) -> None:
+        if not os.path.exists(path):
+            raise ValueError(f"Dataset path '{path}' does not exist.")
+
+        paths = glob.glob(os.path.join(path, "**/*.cypher"), recursive=True)
+        if not paths:
+            raise ValueError(f"No .cypher files found in dataset path '{self.dataset_path}'.")
+
+        # Separate nodes and edges
+        nodes_paths = [p for p in paths if p.endswith("nodes.cypher")]
+        edges_paths = [p for p in paths if p.endswith("edges.cypher")]
+
+        # Process nodes.cypher files first
+        for node_path in nodes_paths:
+            print(f"Start loading dataset from '{node_path}'...")
+            try:
+                with open(node_path, 'r') as file:
+                    data = file.read()
+                    for line in data.splitlines():
+                        self.run_query(line)
+            except Exception as e:
+                print(f"Error loading dataset from '{node_path}': {e}")
+
+        # Process edges.cypher files next
+        for edge_path in edges_paths:
+            print(f"Start loading dataset from '{edge_path}'...")
+            try:
+                with open(edge_path, 'r') as file:
+                    data = file.read()
+                    for line in data.splitlines():
+                        self.run_query(line)
+            except Exception as e:
+                print(f"Error loading dataset from '{edge_path}': {e}")
+
+        print(f"Finished loading {len(nodes_paths)} + {len(edges_paths)} datasets.")
+
+    def run_query(self, query_code):
+        with self.driver.session() as session:
+            results = session.run(query_code[0])
+            result_list = [record for record in results]
+            return result_list
+
+    def query_Generator(self, requests,node_map):
+        if node_map is None:
+            raise Exception('error')
```

Comment:
Please add more information on what error is being raised here

Reply:
added more details about the specific error